src/edu/washington/escience/myria/operator/Split.java

```diff
@@ -0,0 +1,146 @@
+   * @param columnName name of string column to split using regex
+   * @param regex regular expression to split string input column on
+   */
+  public Split(final Operator child, final String columnName, final String regex) {
+    super(child);
+    this.columnName = columnName;
+    pattern = Pattern.compile(regex);
+  }
+
+  @Override
+  protected TupleBatch fetchNextReady() throws DbException {
+    final Operator child = getChild();
+    for (TupleBatch tb = child.nextReady(); tb != null; tb = child.nextReady()) {
+      List<ColumnBuilder<?>> colBuilders = new ArrayList<>(tb.numColumns());
+      for (int i = 0; i < tb.numColumns(); ++i) {
+        if (i != splitColIdx) {
+          colBuilders.add(ColumnFactory.allocateColumn(tb.asColumn(i).getType()));
+        }
+      }
+      final ColumnBuilder<?> splitColBuilder = ColumnFactory.allocateColumn(Type.STRING_TYPE);
+      colBuilders.add(splitColBuilder);
+      for (int rowIdx = 0; rowIdx < tb.numTuples(); ++rowIdx) {
+        try {
+          String colValue = tb.asColumn(splitColIdx).getString(rowIdx);
+          // We must specify a negative value for the limit parameter to avoid discarding trailing empty strings:
+          // http://docs.oracle.com/javase/7/docs/api/java/lang/String.html#split(java.lang.String,%20int)
+          String[] splits = pattern.split(colValue, -1);
+          for (String segment : splits) {
+            splitColBuilder.appendString(segment);
+            // For each split segment, duplicate the values of all other columns in this row.
+            for (int inputColIdx = 0; inputColIdx < inputColIndexToOutputColIndex.length; ++inputColIdx) {
+              int outputColIdx = inputColIndexToOutputColIndex[inputColIdx];
+              if (outputColIdx != -1) {
+                TupleUtils.copyValue(tb.asColumn(inputColIdx), rowIdx, colBuilders.get(outputColIdx));
+              }
+            }
+          }
+        } catch (Exception e) {
+          throw new DbException(e);
+        }
+      }
+      // TODO: this should be factored out into a common utility method
+      List<Column<?>> columns = new ArrayList<>();
+      for (ColumnBuilder<?> colBuilder : colBuilders) {
+        columns.add(colBuilder.build());
+      }
+      return new TupleBatch(outputSchema, columns);
+    }
+    return null;
+  }
+
+  @Override
+  protected void init(final ImmutableMap<String, Object> execEnvVars) throws DbException {
+    final Operator child = getChild();
+    Preconditions.checkNotNull(child, "Child operator cannot be null");
+    final Schema inputSchema = child.getSchema();
+    Preconditions.checkNotNull(inputSchema, "Child schema cannot be null");
+    splitColIdx = inputSchema.columnNameToIndex(columnName);
+    final Type colType = inputSchema.getColumnType(splitColIdx);
+    Preconditions.checkState(colType == Type.STRING_TYPE, "Column to split \"" + columnName
+        + "\" must have type STRING_TYPE but has type " + colType);
```

Comment:
in all the guava functions like `Preconditions`, it is more efficient to write them as follows:

```java
Preconditions.checkState(colType == Type.STRING_TYPE, "Column to split \"%s\" (%s) must have type STRING_TYPE", columnName, colType);
```

This avoids unnecessary string construction.